Sources/Orbit/Components/SegmentedSwitch.swift

```diff
@@ -0,0 +1,157 @@
+                }
+                .background(selectionBackground)
+                .background(roundedBackground)
+                .fixedSize(horizontal: false, vertical: true)
+            }
+        )
+    }
+
+    @ViewBuilder func segment(title: Value) -> some View {
+        Text(title.description)
+            .lineLimit(1)
+            .padding(.vertical, .xSmall)
+            .frame(maxWidth: .infinity)
+            .contentShape(Rectangle())
+            .onTapGesture {
+                selection = title
+            }
+    }
+
+    @ViewBuilder var separator: some View {
+        Rectangle()
+            .fill(borderColor)
+            .frame(width: separatorWidth)
+    }
+
+    @ViewBuilder var selectionBackground: some View {
+        GeometryReader { reader in
+            if selection != nil {
+                Color.whiteDarker
+                    .frame(width: (reader.size.width - 2 * borderWidth) / 2)
+                    .clipShape(RoundedRectangle(cornerRadius: BorderRadius.default))
+                    .padding(borderWidth)
+                    .elevation(.level1)
+                    .offset(computeSelectionOffset(in: reader))
+                    .animation(.easeOut(duration: 0.2), value: selection)
+            }
+        }
+    }
+
+    @ViewBuilder var roundedBackground: some View {
+        RoundedRectangle(cornerRadius: BorderRadius.default)
+            .fill(backgroundColor)
+            .overlay(
+                RoundedRectangle(cornerRadius: BorderRadius.default)
+                    .stroke(borderColor, lineWidth: borderWidth)
+            )
+    }
+
+    private var message: Message? {
+        if case .error = _message, selection != nil {
+            return nil
+        }
+
+        return _message
+    }
+
+    private var backgroundColor: Color {
+        selection != nil ? .cloudNormal : .whiteDarker
+    }
+
+    private var borderColor: Color {
+        guard selection == nil else { return .clear }
+
+        switch message {
+            case .error:
+                return .redNormal
+            default:
+                return .cloudNormal
+        }
+    }
+
+    private func computeSelectionOffset(in reader: GeometryProxy) -> CGSize {
+        .init(width: selection == firstOption ? 0 : reader.size.width / 2 - borderWidth, height: 0)
+    }
+
+    public init(
+        label: String,
+        message: Message? = nil,
+        firstOption: Value,
+        secondOption: Value,
+        selection: Binding<Value?>
+    ) where Value: LosslessStringConvertible {
+        self._message = message
+        self._selection = selection
+        self.firstOption = firstOption
+        self.secondOption = secondOption
+        self.label = label
+    }
+}
+
+// MARK: - Previews
+struct SegmentedSwitchPreviews: PreviewProvider {
+    static var previews: some View {
+        PreviewWrapper {
+            segmentedSwitch()
+                .previewDisplayName("Unselected")
```

Comment:
It is better to extract each in a separate property (and choose one as a snapshot), as you will get a name for free with `previewDisplayName()`.